backend/services/simulation_engine.py

```
@@ -362,6 +389,158 @@ def _format_conversation_for_ai(self, conversation_history: List[Dict[str, str]]
+            # Use LangChain grading agent
+            result = await self.grading_agent.validate_goal_achievement(
+                conversation_history=conversation_text,
+                scene_goal=scene_goal,
+                scene_description=scene_description,
+                current_attempts=current_attempts,
+                max_attempts=max_attempts
+            )
+            
+            return result
+            
+        except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)